homeassistant/components/dsmr/config_flow.py

```diff
@@ -210,6 +228,7 @@ async def async_step_setup_network(
                 vol.Required(CONF_HOST): str,
                 vol.Required(CONF_PORT): int,
                 vol.Required(CONF_DSMR_VERSION): vol.In(DSMR_VERSIONS),
+                vol.Required(CONF_IS_RFXTRX): bool,
```

Comment:
I'd rather use a selection form here for the transport protocol to use. The default should be the standard dsmr protocol and the user can select to change to rfxtrx protocol.

That way, if more protocols are added in the future to dsmr we can just extend this list of protocols.

Reply:
Even better would be if we could autodect the protocol to use.

Reply:
Auto-detect is a bit confusing maybe (in this case)... Currently, this is the first step of DSMR:

![image](https://user-images.githubusercontent.com/195327/150305640-efd35918-69f8-47c6-aac4-be21125c6316.png)

I think adding a third option to that list is the most transparent we can be.

Reply:
Rfxtrx also needs the choice serial or network right?

Reply:
Oh, right. In that case, auto-detecting sounds like a better option.

Reply:
Looking for some guidance here: how would I go about auto-detecting this?
Just give it a try with the regular DSMR protocol and it that does not work (times out?) try again with the RFXtrxDSMR protocol?

Reply:
That would indeed be the approach

Reply:
Updated the code to use auto-detection instead of a boolean config item.